src/makeInfinityAdder.js

```diff
@@ -6,7 +6,22 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const makeAdder = function(num) {
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+    sum += num;
+
+    return makeAdder;
+  };
+
+  return makeAdder;
```

Comment:
```suggestion
```

You can write `return function makeAdder(num) {` in the line 11 and return it right away